.github/workflows/tests.yml

```diff
@@ -0,0 +1,155 @@
+  #         #
+  #         # but browser-actions/setup-chrome uses different names for the binaries:
+  #         # https://github.com/browser-actions/setup-chrome#usage
+  #         #
+  #         # Specify version to avoid mismatch with `chrome` binary.
+  #         chromedriver-version: ${{ env.chromedriver_version }}
+  #     - uses: actions/setup-node@v3
+  #       with:
+  #         node-version: lts/*
+  #     - run: |
+  #         chromedriver --url-base=/wd/hub &
+  #         sudo Xvfb -ac :99 -screen 0 1280x1024x24 > /dev/null 2>&1 & # optional
+  #
+  #     - name: Install dependencies
+  #       run: npm install
+  #
+  #     - name: Install sf CLI
+  #       run: npm install --global @salesforce/cli
+  #
+  #     - name: Setup scratch org
+  #       env:
+  #         SF_HUB_USERNAME: ${{ secrets.SF_HUB_USERNAME }}
+  #         SF_OAUTH2_CLIENT_ID: ${{ secrets.SF_OAUTH2_CLIENT_ID }}
+  #         SF_OAUTH2_JWT_KEY: ${{ secrets.SF_OAUTH2_JWT_KEY }}
+  #       run: |
+  #         npx zx scripts/org-setup.mjs
+  #
+  #     - name: Run browser tests
+  #       run: |
+  #         SF_USERNAME=$(sf org display user --target-org jsforce-test-org --json | jq -r '.result.username') \
+  #         SF_PASSWORD=$(sf org display user --target-org jsforce-test-org --json | jq -r '.result.password') \
+  #         SF_LOGIN_URL=$(sf org display user --target-org jsforce-test-org --json | jq -r '.result.instanceUrl') \
+  #         DISPLAY=:99 CHROME_BIN=$(which chrome) \
+  #         npm run test:browser-ci
+
+  test-node-linux:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node_version: [lts/-1, lts/*, latest]
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node_version }}
+  
+      - name: Install dependencies
+        run: npm install
+  
+      - name: Lint and Type Check
+        run: |
+          npm run lint
+          npm run build:types
+          npm run typecheck
+  
+      - name: Install sf CLI
+        run: npm install --global @salesforce/cli
+  
+      - name: Setup scratch org
+        env:
+          SF_HUB_USERNAME: ${{ secrets.SF_HUB_USERNAME }}
+          SF_OAUTH2_CLIENT_ID: ${{ secrets.SF_OAUTH2_CLIENT_ID }}
+          SF_OAUTH2_JWT_KEY: ${{ secrets.SF_OAUTH2_JWT_KEY }}
+        run: |
+          npx zx scripts/org-setup.mjs
+  
+      - name: Run Node tests
+        env:
+          SF_OAUTH2_CLIENT_ID: ${{ secrets.SF_OAUTH2_CLIENT_ID }}
+          SF_OAUTH2_CLIENT_SECRET: ${{ secrets.SF_OAUTH2_CLIENT_SECRET }}
+        run: |
+          SF_USERNAME=$(sf org display user --target-org jsforce-test-org --json | jq -r '.result.username') \
+          SF_PASSWORD=$(sf org display user --target-org jsforce-test-org --json | jq -r '.result.password') \
+          SF_LOGIN_URL=$(sf org display user --target-org jsforce-test-org --json | jq -r '.result.instanceUrl') \
+          npm run test:node
```

Comment:
inline env vars with scratch org credentials (see the backslash at the EOL) to avoid exposing them in workflow logs.
We could also do `echo "SF_USERNAME=$(sf org display user --target-org jsforce-test-org --json | jq -r '.result.username')" >> $GITHUB_ENV` but since they aren't repo secrets they will not be redacted in this section: 
![Screenshot 2023-08-01 at 11 09 33](https://github.com/jsforce/jsforce/assets/6853656/aa143de2-6164-4d9d-bd70-e90427d99b68)